text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,863 @@
+
+### Why don't we download schema url contents?
+
+OpenTelemetry needs to work in environments that have no/limited access to the external internet. We entertained, and
+dismissed merging solutions that *require* access to the contents of `schema_url` to work. While the core algorithm
+*cannot require* this access, we *should* be able to provide improved processing and algorithms that may leverage this data.
+
+For example:
+
+- Within an SDK, we can registry entity schema information with `EntityDetector`.
+- The OpenTelemetry Collector can allow registered `schema_url` via configuration
+  or (optionally) download schema on demand.
+
+This design does not prevent these solutions, but provides the baseline/fallback
+where `schema_url` is not accessible and entities must still be usable.
+
+## Prior art and alternatives
+
+Previously, we have a few unaccepted oteps, e.g. ([OTEP 208](https://github.com/open-telemetry/oteps/pull/208)).  Additionally, there are some alternatives that were considered in the Entities WG and rejected.
+
+Below is a brief discussion of some design decisions:
+
+- **Only associating one entity with a Resource.**  This was rejected, as too high a friction point in evolving semantic conventions and allowing independent systems to coordinate identity + entities within the OpenTelemetry ecosystem.  Eventually, this would force OpenTelemetry to model all possibly entities in the world and understand their interaction or otherwise prevent non-OpenTelemetry instrumentation from interacting with OpenTelemetry entities.
+- **Embed fully Entity in Resource.** This was rejected because it makes it easy/trivial for Resource attributes and Entities to diverge.  This would prevent the backwards/forwards compatibility goals and also require all participating OTLP users to leverage entities. Entity should be an opt-in / additional feature that may or may not be engaged with, depending on user need.
+- **Re-using resource detection as-is** This was rejected as not having a viable compatibility path forward.  Creating a new set of components that can preserve existing behavior while allowing users to adopt the new functionality means that users have better control of when they see / change system behavior, and adoption is more obvious across the ecosystem.
+
+## Future Posibilities
+
+This proposal opens the door for addressing issues where an Entity's lifetime does not match an SDK's lifetime, in addition to providing a data model where mutable (descriptive) attributes can be changed over the lifetime of a resource without affecting its identity.  We expect a follow-on OTEP which directly handles this issue.
```

Comment:
Two things I would see valuable to be addressed going forward as well:

* (Optional) attributes per resource/entity to identify where it was set. This is especially useful in the cases described above where multiple places in the flow try to set a resource/entity and one wins over the other. Imagine that I configure a collector with a processor that should set some k8s.pod.id, but for whatever reason the id is not what I expect, because I am not aware that the id is set by the k8s operator via an environment variable. Debugging that would become a mess, so having an (opt in?) attribute that annotates where the data is coming from would be helpful.
* Maybe as part of the semantic conventions a definition on how certain attributes should be collected. E.g. we had a discussion in the past around [`host.id`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/host/), what the value should be and where it should come from.

Reply:
(2) is already called out and on the task queue for Semantic Conventions + Entities WG.

(1) Is an interesting idea. I think there will be a whole "debugability" thing to figure out, but honestly the scenario you mention is already an issue today.  It's an interesting idea to call out as a potential solution or "diagnostic" mode we could support.